packages/workflow/src/result.ts

```diff
@@ -11,3 +13,18 @@ export const createResultError = <E = unknown>(error: E): ResultError<E> => ({
 	ok: false,
 	error,
 });
+
+/**
+ * Executes the given function and converts it to a Result object.
+ *
+ * @example
+ * const result = toResult(() => fs.writeFileSync('file.txt', 'Hello, World!'));
+ */
+export const toResult = <T, E extends Error = Error>(fn: () => T): Result<T, E> => {
```

Comment:
nice. I did not realize that we had already added a `Result` type. Thanks.
but, since this is not used in editor or nodes (and since we don't know if those teams want to use this), can we please move this to core as well. We can do that in another PR though.
Also, we should add tests for all this code.